source/server/server.cc

```diff
@@ -175,6 +172,18 @@ void InstanceImpl::initialize(Options& options,
     Json::ObjectSharedPtr config_json = Json::Factory::loadFromFile(options.configPath());
     Config::BootstrapJson::translateBootstrap(*config_json, bootstrap);
   }
+
+  // Needs to happen as early as possible in the instantiation to preempt the objects that require
+  // stats.
+  tag_extractors_ = Config::Utility::createTagExtractors(bootstrap);
+  stats_store_.setTagExtractors(tag_extractors_);
+
+  server_stats_.reset(
+      new ServerStats{ALL_SERVER_STATS(POOL_GAUGE_PREFIX(stats_store_, "server."))});
+
+  failHealthcheck(false);
```

Comment:
How come this moved?

Reply:
This method calls `server_stats_->live_.set(!fail);`, so `server_stats_` needs to be initialized before it's called.